File: tyama_icpc2007rA.scala
//usr/bin/env scala $0 $@;exit

object icpc2007rA{
	def main(args: Array[String]){
		val cin=new java.util.Scanner(System.in)
		var n=cin.nextInt()
		while(n>0){
			val k=cin.nextInt()
			val m=cin.nextInt()
			var r=0
			for(i<-1 to n-1)r=(r+k)%i
			println((r+m)%n+1)
			n=cin.nextInt()
		}
	}
}

File: tyama_icpc2007rA.swift
//usr/bin/env swift $0 $@;exit
import CoreFoundation
var n=0,k=0,m=0,r=0,i=0
_=withUnsafeMutablePointer(to:&n){withVaList([OpaquePointer($0)]){vscanf("%d",$0)}}
while n>0 {
	_=withUnsafeMutablePointer(to:&k){withVaList([OpaquePointer($0)]){vscanf("%d",$0)}}
	_=withUnsafeMutablePointer(to:&m){withVaList([OpaquePointer($0)]){vscanf("%d",$0)}}
	r=0
	for i in 1..<n {r+=k;r%=i}
	print((r+m)%n+1)
	_=withUnsafeMutablePointer(to:&n){withVaList([OpaquePointer($0)]){vscanf("%d",$0)}}
}
